README.md

```diff
@@ -55,10 +55,120 @@ console.log(hi); // => howdy world!
 
 # Publishing Binaries
 
-It's a good idea to publish pre-built binaries of your module if you want others to be able to install it properly on their own systems, which can very likely not have a compiler like `gcc` or `clang`. Node-pre-gyp does a lot of the heavy lifting for us (like detecting which system you are building on and deploying to s3) but you'll need a few things configured to get started.
+It's a good idea to publish pre-built binaries of your module if you want others to be able to easily install it on their system without needing to install a compiler like g++ or clang++. Node-pre-gyp does a lot of the heavy lifting for us (like detecting which system you are building on and deploying to s3) but you'll need a few things configured to get started.
+
+1) In the `package.json`, update the `"binary"` field to the appropriate s3 bucket `host`.
+
+For Mapbox staff we recommend using a host setting of `"host": "https://mapbox-node-binary.s3.amazonaws.com",` which will publish to `s3://mapbox-node-binary/<your module name>`.
+
+Note: for namespaced modules the path will end up being `s3://mapbox-node-binary/@org/<your module name>`.
+
+2) Copy the ci.template.js
+
+Copy the `ci.template.js` from this repo into your repo and place it at  `./cloudformation/ci.template.js`
+
+3) Install deps
+
+```
+npm install @mapbox/cfn-config
+npm install @mapbox/cloudfriend
+```
+
+4) Create a user with permissions to upload to `s3://mapbox-node-binary/<your module name>/`
+
+First configure your AWS creds. You will need to set at least the `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` keys. And also `AWS_SESSION_TOKEN` if you are using 2-factor auth.
+
+Then run:
+
+```bash
+./node_modules/.bin/validate-template cloudformation/ci.template.js
```

Comment:
> write the template without cloudfriend

The benefit of `cloudfriend` is the ability to dynamically grab the `package.json`'s `name` property. This is critical so that the permissions are scoped to the specific module and not the root directory. Fewer deps is better, but not sure how to do this without cloudfriend. Can you think of a way to do this without `cloudfriend`?

> include the dependency

My reasoning in leaving this off (and only mentioning to install them in the `step 3` above) was that this is a one-time thing. But if you think it would be better to add to devDeps, I'm 👍